src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -285,8 +288,14 @@ public String getUserId() {
     	return definitions;
     }
 
+    public void doRebuild(StaplerRequest req, StaplerResponse rsp) throws IOException {
+        throw HttpResponses.error(404, "Promotions may not be rebuilt directly");
```

Comment:
A bit hackish, but ok. Would it be possible to create a WebClient-based test for it?

Reply:
Yeah it's definitely hacky, I'll add a test. It mirrors `doBuild` in [PromotionProcess](https://github.com/dwnusbaum/promoted-builds-plugin/blob/ee60da19a57a496a3ba5662146a2e2a3341ff2e9/src/main/java/hudson/plugins/promoted_builds/PromotionProcess.java#L533-L535). New promotions will use `PromotionRebuildValidator` so that a `RebuildAction` never gets added, so this should only affect already existing promotions.